gcloud.go

```diff
@@ -106,55 +142,60 @@ func createVMs(names []string, opts VMOpts) error {
 	}
 	defer os.Remove(filename)
 
-	// Fixed args.
-	args := []string{
-		"compute", "instances", "create",
-		"--subnet", "default",
-		"--maintenance-policy", "MIGRATE",
-		"--service-account", "21965078311-compute@developer.gserviceaccount.com",
-		"--scopes", "https://www.googleapis.com/auth/devstorage.read_only,https://www.googleapis.com/auth/logging.write,https://www.googleapis.com/auth/monitoring.write,https://www.googleapis.com/auth/servicecontrol,https://www.googleapis.com/auth/service.management.readonly,https://www.googleapis.com/auth/trace.append",
-		"--image", "ubuntu-1604-xenial-v20171002",
-		"--image-project", "ubuntu-os-cloud",
-		"--boot-disk-size", "10",
-		"--boot-disk-type", "pd-ssd",
-	}
+	for i, name := range names {
+		// Fixed args.
+		args := []string{
+			"compute", "instances", "create",
+			"--subnet", "default",
+			"--maintenance-policy", "MIGRATE",
+			"--service-account", "21965078311-compute@developer.gserviceaccount.com",
+			"--scopes", "https://www.googleapis.com/auth/devstorage.read_only,https://www.googleapis.com/auth/logging.write,https://www.googleapis.com/auth/monitoring.write,https://www.googleapis.com/auth/servicecontrol,https://www.googleapis.com/auth/service.management.readonly,https://www.googleapis.com/auth/trace.append",
+			"--image", "ubuntu-1604-xenial-v20171002",
+			"--image-project", "ubuntu-os-cloud",
+			"--boot-disk-size", "10",
+			"--boot-disk-type", "pd-ssd",
+		}
 
-	// Dynamic args.
-	if opts.UseLocalSSD {
-		args = append(args, "--local-ssd", "interface=SCSI")
+		// Dynamic args.
+		if opts.UseLocalSSD {
+			args = append(args, "--local-ssd", "interface=SCSI")
+		}
+		args = append(args, "--machine-type", opts.MachineType)
+		args = append(args, "--labels", fmt.Sprintf("lifetime=%s", opts.Lifetime))
+
+		args = append(args, "--metadata-from-file", fmt.Sprintf("startup-script=%s", filename))
+		args = append(args, "--project", project)
+		args = append(args, "--zone", zones[i%len((zones))])
+		//args = append(args, "--additional-zones",strings.Join(zones[1:],","))
+		args = append(args, name)
+		cmd := exec.Command("gcloud", args...)
```

Comment:
this is going to be much slower. How about grouping the commands by zone and providing the list of all VMs in that zone? An alternative would be to run commands in goroutines, but what has a risk of making too many gcloud calls.

Reply:
I vote goroutines; I think the common case (to start) will be spinning up _n_ nodes in _n_ regions.

Reply:
Well, the common case for geo-distributed. But if you're running all nodes in one region, why not let `gcloud` handle that for you? I haven't tried 50 concurrent instances of gcloud, but azure would melt your computer.